File: fe-antd-v1/src/model/enum/SysOtherAppTypeEnum.ts
import {IEnum} from "@/model/enum/CommonEnum";
import {ProSchemaValueEnumType} from "@ant-design/pro-components";

export interface ISysOtherAppTypeEnum {

    WX_MINI_PROGRAM: IEnum<number>,

    WX_OFFICIAL_ACCOUNT: IEnum<number>,

    WX_WORK: IEnum<number>,

    ALI_PAY_PROGRAM: IEnum<number>,

    BAI_DU: IEnum<number>,

    VOLC_ENGINE: IEnum<number>,

}

// 第三方应用类型，枚举类
export const SysOtherAppTypeEnum: ISysOtherAppTypeEnum = {

    WX_MINI_PROGRAM: {
        code: 101,
        name: '微信小程序',
    },

    WX_OFFICIAL_ACCOUNT: {
        code: 102,
        name: '微信公众号',
    },

    WX_WORK: {
        code: 103,
        name: '企业微信',
    },

    ALI_PAY_PROGRAM: {
        code: 201,
        name: '支付宝小程序',
    },

    BAI_DU: {
        code: 301,
        name: '百度',
    },

    VOLC_ENGINE: {
        code: 401,
        name: '火山引擎',
    },

}

export const SysOtherAppTypeEnumDict = new Map<number, ProSchemaValueEnumType>();

Object.keys(SysOtherAppTypeEnum).forEach(key => {

    const item = SysOtherAppTypeEnum[key] as IEnum<number>;

    SysOtherAppTypeEnumDict.set(item.code!, {text: item.name})

})

File: be-engine-v1/be-other-app/src/main/java/com/cmcorg20230301/be/engine/other/app/model/enums/SysOtherAppTypeEnum.java
package com.cmcorg20230301.be.engine.other.app.model.enums;

import com.baomidou.mybatisplus.annotation.EnumValue;
import com.cmcorg20230301.be.engine.other.app.model.interfaces.ISysOtherAppType;
import com.fasterxml.jackson.annotation.JsonValue;

import lombok.AllArgsConstructor;
import lombok.Getter;

/**
 * 第三方应用类型，枚举类
 */
@AllArgsConstructor
@Getter
public enum SysOtherAppTypeEnum implements ISysOtherAppType {

    WX_MINI_PROGRAM(101), // 微信小程序

    WX_OFFICIAL_ACCOUNT(102), // 微信公众号

    WX_WORK(103), // 企业微信

    ALI_PAY_PROGRAM(201), // 支付宝小程序

    BAI_DU(301), // 百度

    VOLC_ENGINE(401), // 火山引擎

    MICROSOFT(501), // 微软

    ;

    @EnumValue
    @JsonValue
    private final int code;

}
